node-template.yaml

```diff
@@ -1,36 +1,32 @@
 # Required Parameter: node
 
-- name: orchestrate {{ node.name }} - clean
-  ansible.builtin.file:
-    path: "{{ node.name }}"
-    state: absent
+- name: orchestrate {{ node.name }} - clean and copy templates files to {{ node.name }}
+  ansible.builtin.shell: |
+    rm -rf "{{ node.name }}"
+    mkdir -p "{{ node.name }}"
+    cp -R templates/hadoop-common/* "{{ node.name }}/" 2>/dev/null || true
+    cp -R templates/{{ node.group }}/* "{{ node.name }}/" 2>/dev/null || true
+  args:
+    executable: /bin/bash
+    warn: false
+  changed_when: false
 
-- name: orchestrate {{ node.name }} - hadoop-common - mkdir
-  ansible.builtin.file:
-    path: "{{ node.name }}/{{ item.path }}"
-    state: directory
-  with_community.general.filetree: "templates/hadoop-common"
-  when: item.state == "directory"
```

Comment:
this is definitely the most costly step, in my experiment, it costs 15.94s, so collecting and dedup the dirs first and only creating the unique dirs should be a good idea